### ISSUE TITLE
Confusing message: No build tasks configured

### ISSUE THREAD

**Issue by BurtHarris** [Microsoft/vscode#7877]
- VSCode Version: 1.2.1
- OS Version: Windows 10
- Node.js version 4.4.5 (LTS) version probably not critical...

Steps to Reproduce:

1.   Setup Node.js, then install the grunt-cli globally from a command prompt:   `npm install -G grunt-cli`

2.  Install source files from attached zip into an empty directory.  [junk.zip](https://github.com/Microsoft/vscode/files/322685/junk.zip)

3.  Then from command prompt in that directory, install the dev dependencies with: `npm install`

4. Start VSCode with `code .`

5. From VSCode, Press **Ctrl+Alt+B**.   VSCode correctly says *No build task configured.*  Choose "Configure Task Runner" and choose "Grunt" from the menu of templates.   Save it with **Ctrl+S**.

6. From VSCode, Press **Ctrl+Alt+B** again...

Observed:  

-  VSCode again says *No build task configured*, after we just created the suggested one... 

-  No output in tasks pane 

-  Pressing the "Configure Task Runner" button does nothing

-  Contents of URL referenced in the generated file are too complex to really help...

Expected:

- Should have run grunt, producing results in the output pane

**Comment by BurtHarris** [comment]
Here's the generated tasks.json generated by VSCode 1.2.1:
```
{
    // See https://go.microsoft.com/fwlink/?LinkId=733558
    // for the documentation about the tasks.json format
    "version": "0.1.0",
    "command": "grunt",
    "isShellCommand": true,
    "tasks": []
}
```

**Comment by BurtHarris** [comment]
Possible dupe of #7881

**Comment by BurtHarris** [comment]
Note however, that the **No build tasks configured.** message can be generated for a number of different reasons, e.g. a JSON syntax error in tasks.json.   Better explanation in this message would be valuable.

**Comment by dbaeumer** [comment]
The problem is that the task runner doesn't detect any grunt task that looks like a build tasks. Adding 
```json
        {
            "taskName": "default",
            "isBuildCommand": true
        }
```
Fixes the problem. Should look into a better error message. On top we should make the default task the build task if no better is found.

**Comment by bitcrazed** [comment]
@dbaeumer et. al. - note that the ```"isBuildCommand" param is now deprecated, and that a group should be created instead:

```javascript
    "group": {
        "kind": "build",
        "isDefault": true
    }
```

Proposing a fix to the template in #35352.
